src/main/java/scala_maven/ScalaMojoSupport.java

```diff
@@ -783,47 +790,81 @@ private String getToolClasspath() throws Exception {
     return options;
   }
 
+  private String getScala3ArtifactId(String a) {
+    String _scalaVersionNStr = _scalaVersionN.toString();
+    // int indexOf_ = _scalaVersionNStr.indexOf("-");
+    return _scalaVersionN.major == 3 ? a + "_" + _scalaVersionNStr : a;
+  }
+
   protected File getLibraryJar() throws Exception {
+    String scalaLibrary =
+        _scalaVersionN.major == 3
+            ? getScala3ArtifactId(SCALA3_LIBRARY_ARTIFACTID)
+            : SCALA_LIBRARY_ARTIFACTID;
     if (StringUtils.isNotEmpty(scalaHome)) {
       File lib = new File(scalaHome, "lib");
-      return new File(lib, SCALA_LIBRARY_ARTIFACTID + ".jar");
+      return new File(lib, scalaLibrary + ".jar");
     }
-    return getArtifactJar(
-        getScalaOrganization(), SCALA_LIBRARY_ARTIFACTID, findScalaVersion().toString());
+    return getArtifactJar(getScalaOrganization(), scalaLibrary, findScalaVersion().toString());
```

Comment:
I guess the value of `scalaLibrary` is not right one for scala 3, because `getScala3ArtifactId` will return an artifact name with a version number => not found